### PR TITLE
Add "find a match" button and link it to the matchmaker

### PR DESCRIPTION
**Pull request by adamsmlsn**
The button is located at the bottom of the user's profile.

### CODE REVIEW

**Comment by wangding-google** [comment]
> Thanks for your pull request. It looks like this may be your first contribution to a Google open source project (if not, look below for help). Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> 📝 **Please visit https://cla.developers.google.com/ to sign.**
> 
> Once you've signed (or fixed any issues), please reply here with `@googlebot I signed it!` and we'll verify it.
> 
> #### What to do if you already signed the CLA
> ##### Individual signers
> * It's possible we don't have your GitHub username or you're using a different email address on your commit. Check [your existing CLA data](https://cla.developers.google.com/clas) and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> 
> ##### Corporate signers
> * Your company has a Point of Contact who decides which employees are authorized to participate. Ask your POC to be added to the group of authorized contributors. If you don't know who your Point of Contact is, direct the Google project maintainer to [go/cla#troubleshoot](http://go/cla#troubleshoot) ([Public version](https://opensource.google/docs/cla/#troubleshoot)).
> * The email used to register you as an authorized contributor must be the email used for the Git commit. Check [your existing CLA data](https://cla.developers.google.com/clas) and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> * The email used to register you as an authorized contributor must also be [attached to your GitHub account](https://github.com/settings/emails).
> 
> ℹ️ **Googlers: [Go here](https://goto.google.com/prinfo/https%3A%2F%2Fgithub.com%2Fgoogleinterns%2Fstep21-2020%2Fpull%2F12) for more info**.

Hi Adam please resolve this? Thanks!

File: step/src/main/java/servlets/MatchingServlet.java [hunk 1]
```diff
@@ -23,9 +29,25 @@
 @WebServlet("/matching")
 public class MatchingServlet extends HttpServlet {
 
+  public static String REQUEST_TYPE = "requestType";
+  public static String REQUEST_TYPE_MATCH = "match-request";
```
Comment (lailagamaleldin) [inline step/src/main/java/servlets/MatchingServlet.java]: nit: it might be better style to use either camel-case of dash-separated words for both strings

**Review by lailagamaleldin — APPROVED** [review]
LGTM

**Review by nganttran — APPROVED** [review]

File: step/src/main/java/servlets/MatchingServlet.java [hunk 2]
```diff
@@ -23,9 +29,25 @@
...
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // TODO: implement
   }
 
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String requestType = request.getParameter(REQUEST_TYPE);
+    if(requestType.equals(REQUEST_TYPE_MATCH)) {
+      UserService userService = UserServiceFactory.getUserService();
+      com.google.appengine.api.users.User userServiceUser = userService.getCurrentUser();
```
Comment (nganttran) [inline step/src/main/java/servlets/MatchingServlet.java:45]: Can you import the API as a library above the class name?

**Comment by adamsmlsn** [comment]
@googlebot I signed it!

File: step/src/main/java/servlets/MatchingServlet.java (hunk 1 above)
Comment (adamsmlsn) [reply step/src/main/java/servlets/MatchingServlet.java]: Yeah I was debating myself about this. I saw requestType more as a variable name (so camel case) but I'll go ahead and make it consistent

File: step/src/main/java/servlets/MatchingServlet.java (hunk 2 above)
Comment (adamsmlsn) [reply step/src/main/java/servlets/MatchingServlet.java:45]: It's already imported but I chose to use the full class name to reduce confusion (for the compiler and programmer) about which User object is the API's and which is ours.

**Pull request merged**